plugins/asus-hid/fu-asus-hid-device.c

```diff
@@ -0,0 +1,586 @@
+/*
+ * Copyright 2024 Mario Limonciello <superm1@gmail.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1-or-later
+ */
+
+#include "config.h"
+
+#include "fu-asus-hid-child-device.h"
+#include "fu-asus-hid-device.h"
+#include "fu-asus-hid-struct.h"
+
+struct _FuAsusHidDevice {
+	FuHidDevice parent_instance;
+	guint8 num_mcu;
+};
+
+G_DEFINE_TYPE(FuAsusHidDevice, fu_asus_hid_device, FU_TYPE_HID_DEVICE)
+
+#define FU_ASUS_HID_DEVICE_TIMEOUT 200 /* ms */
+
+static gboolean
+fu_asus_hid_device_transfer_feature(FuDevice *device,
+				    GByteArray *req,
+				    GByteArray *res,
+				    guint8 report,
+				    GError **error)
+{
+	FuHidDevice *hid_dev;
+
+	/* can be called from child or proxy */
+	if (FU_IS_HID_DEVICE(device))
+		hid_dev = FU_HID_DEVICE(device);
+	else
+		hid_dev = FU_HID_DEVICE(fu_device_get_proxy(device));
+
+	if (req != NULL) {
+		if (!fu_hid_device_set_report(hid_dev,
+					      report,
+					      req->data,
+					      req->len,
+					      FU_ASUS_HID_DEVICE_TIMEOUT,
+					      FU_HID_DEVICE_FLAG_IS_FEATURE,
+					      error)) {
+			g_prefix_error(error, "failed to send packet: ");
+			return FALSE;
+		}
+	}
+	if (res != NULL) {
+		if (!fu_hid_device_get_report(hid_dev,
+					      report,
+					      res->data,
+					      res->len,
+					      FU_ASUS_HID_DEVICE_TIMEOUT,
+					      FU_HID_DEVICE_FLAG_IS_FEATURE,
+					      error)) {
+			g_prefix_error(error, "failed to receive packet: ");
+			return FALSE;
+		}
+	}
+
+	return TRUE;
+}
+
+static gboolean
+fu_asus_hid_device_init_seq(FuDevice *device, GError **error)
+{
+	g_autoptr(GByteArray) cmd = fu_struct_asus_hid_command_new();
```

Comment:
I'd much rather you use the `FuStructAsusHidCommand` typedef here -- then the compiler will do type checking.